src/main/java/uk/gov/register/presentation/ArchiveCreator.java

```diff
@@ -1,77 +1,78 @@
 package uk.gov.register.presentation;
 
 import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import io.dropwizard.jackson.Jackson;
+import uk.gov.register.presentation.dao.Record;
 
 import javax.ws.rs.core.StreamingOutput;
+import java.io.Closeable;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.util.HashSet;
+import java.io.OutputStream;
 import java.util.List;
-import java.util.Set;
+import java.util.stream.Collectors;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 
 public class ArchiveCreator {
-    public StreamingOutput create(RegisterDetail registerDetail, List<DbEntry> entries) {
+
+    public StreamingOutput create(RegisterDetail registerDetail, List<Record> records) {
         return output -> {
-            ZipOutputStream zos = new ZipOutputStream(output);
+            try (ZipEntryWriter zipEntryWriter = new ZipEntryWriter(output)) {
+
+                zipEntryWriter.writeEntry("register.json", registerDetail);
 
-            ObjectMapper om = new ObjectMapper();
-            om.configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false);
+                records.stream()
+                        .collect(Collectors.toMap(e -> e.entry.getSha256hex(), e -> e))
```

Comment:
If there are any entries in this list of records that refer to the same item, this will raise a "duplicate key" exception since e.entry.getSha256hex() will be the same. Should the create() method take separate lists of Entry and Item objects to avoid this?